Source/WebCore/rendering/RenderThemeIOS.mm

```diff
@@ -1203,6 +1203,15 @@ static void adjustInputElementButtonStyle(RenderStyle& style, const HTMLInputEle
     if (!element)
         return;
 
+    auto styleColorOptions = element->document().styleColorOptions(&style);
+    auto borderColor = systemColor(CSSValueWebkitControlBackground, styleColorOptions);
+    style.setBorderTopColor(borderColor);
+    style.setBorderBottomColor(borderColor);
+    style.setBorderLeftColor(borderColor);
+    style.setBorderRightColor(borderColor);
+
+    style.setBackgroundColor(systemColor(CSSValueAppleSystemOpaqueTertiaryFill, styleColorOptions));
```

Comment:
As discussed offline with Tim, this causes tests to fail (and breaks behavior) as it limits author customizability of native search fields on iOS. While we don't offer the same level of customization on macOS, due to AppKit limitations, we should not regress iOS behavior.

I think we will need to introduce `hasAuthorExplicitlySet` bits on `RenderStyle` (similar to the existing `hasExplicitlySet` bits, but specifically if the value is set in author style sheets) to decide when to applying this forcing of values.

We had discussed possibly painting the search field appearance ourselves, but that does not solve the custom border/background issue.

There is also the option of dropping native appearance entirely when a custom border/background is set, but that would reduce the level of customization available for search fields today.